jol-core/src/main/java/org/openjdk/jol/info/ClassLayout.java

```diff
@@ -114,24 +127,33 @@ public static ClassLayout parseInstance(Object instance, Layouter layouter) {
      * @param instanceSize instance size
      * @param check        whether to check important invariants
      */
-    public ClassLayout(ClassData classData, SortedSet<FieldLayout> fields, int headerSize, long instanceSize, boolean check) {
-        this.classData = classData;
-        this.fields = fields;
-        this.headerSize = headerSize;
-        this.size = instanceSize;
+    public static ClassLayout createClassLayout(ClassData classData, SortedSet<FieldLayout> fields, int headerSize, long instanceSize, boolean check) {
         if (check) {
-            checkInvariants();
+            checkInvariants(fields, instanceSize);
         }
+        // calculate loses
+        long nextFree = headerSize;
+        long lossesInternal = 0;
+        for (FieldLayout fieldLayout : fields) {
+            if (fieldLayout.offset() > nextFree) {
+                long fieldLayoutSize = fieldLayout.offset() - nextFree;
+                lossesInternal += fieldLayoutSize;
+            }
+            nextFree = fieldLayout.offset() + fieldLayout.size();
+        }
+        long lossesExternal = instanceSize != nextFree ? instanceSize - nextFree : 0;
+        long lossesTotal = lossesInternal + lossesExternal;
```

Comment:
I think this whole block would be more readable if we shorten the variable names.

```
        long next = headerSize;
        long internal = 0;
        for (FieldLayout fl : fields) {
            if (fl.offset() > nextFree) {
                internal += fl.offset() - nextFree;
            }
            next = fl.offset() + fl.size();
        }
        long external = (instanceSize != next) ? (instanceSize - next) : 0;
        long total = internal + external;
```

Reply:
ok, I just kept original variable names for more clear history